cert/ca.go

```diff
@@ -46,6 +50,8 @@ func NewCAPoolFromBytes(caPEMs []byte) (*NebulaCAPool, error) {
 
 	if expired {
 		return pool, ErrExpired
+	} else if caTooNew {
```

Comment:
Kind of a wild situation but we could have an expired and a too new cert and we wouldn't know about the too new one until the expired one was removed.

Reply:
I was thinking maybe we hold back one of the errors and if the other appears then we wrap that. Then we can check if the error appears in the chain instead of direct equality. 

It would be a bummer to log from this here IMO

Reply:
@wadey suggested `(pool CAPool, warnings []error, err error)` so I gave that a try. It feels pretty nice.